okhttp/src/main/java/io/grpc/okhttp/OkHttpProtocolNegotiator.java

```diff
@@ -133,6 +140,59 @@ public String getSelectedProtocol(SSLSocket socket) {
     private static final OptionalMethod<Socket> SET_NPN_PROTOCOLS =
         new OptionalMethod<>(null, "setNpnProtocols", byte[].class);
 
+    // Available on Android 10.0+.
+    // SSLParameters#setApplicationProtocols(String[])
+    private static final OptionalMethod<SSLParameters> SET_APPLICATION_PROTOCOLS =
+        new OptionalMethod<>(null, "setApplicationProtocols", String[].class);
+    // SSLParameters#getApplicationProtocols()
+    private static final OptionalMethod<SSLParameters> GET_APPLICATION_PROTOCOLS =
+        new OptionalMethod<>(String[].class, "getApplicationProtocols");
+    // SSLSocket#getApplicationProtocol()
+    private static final OptionalMethod<SSLSocket> GET_APPLICATION_PROTOCOL =
+        new OptionalMethod<>(String.class, "getApplicationProtocol");
+
+    // Available on Android 7.0+.
+    // SSLParameters#setServerNames(List<SNIServerName>)
+    private static final OptionalMethod<SSLParameters> SET_SERVER_NAMES =
+        new OptionalMethod<>(null, "setServerNames", List.class);
+
+    // Non-null on Android 10.0+.
+    // SSLSockets#isSupportedSocket(SSLSocket)
+    private static final Method SSL_SOCKETS_IS_SUPPORTED_SOCKET;
+    // SSLSockets#setUseSessionTickets(SSLSocket, boolean)
+    private static final Method SSL_SOCKETS_SET_USE_SESSION_TICKET;
+
+    // Non-null on Android 7.0+.
+    // SNIHostName(String)
+    private static final Constructor<?> SNI_HOST_NAME;
+
+    static {
+      // Attempt to find Android 10.0+ APIs.
+      Method sslSocketsIsSupportedSocketMethod = null;
+      Method sslSocketsSetUseSessionTicketsMethod = null;
+      try {
+        Class<?> sslSockets = Class.forName("android.net.ssl.SSLSockets");
+        sslSocketsIsSupportedSocketMethod =
+            sslSockets.getMethod("isSupportedSocket", SSLSocket.class);
+        sslSocketsSetUseSessionTicketsMethod =
+            sslSockets.getMethod("setUseSessionTickets", SSLSocket.class, boolean.class);
+      } catch (Throwable e) {
```

Comment:
Is there a need to catch Throwable instead of Exception? I'd honestly feel best catching just the reflection exceptions.

Reply:
Fixed.